mne/viz/utils.py

```diff
@@ -908,18 +908,22 @@ def _mouse_click(event, params):
         return
     if event.button == 3:
         if params['fig_annotation'] is None:
-            return
-        raw = params['raw']
-        if np.any([c.contains(event)[0] for c in params['ax'].collections]):
-            xdata = event.xdata - params['first_time']
-            onset = _sync_onset(raw, raw.annotations.onset)
-            ends = onset + raw.annotations.duration
-            ann_idx = np.where((xdata > onset) & (xdata < ends))[0]
-            raw.annotations.delete(ann_idx)  # only first one deleted
-        _remove_segment_line(params)
-        _plot_annotations(raw, params)
-        params['plot_fun']()
-        return
+            if not isinstance(event.xdata, tuple):
```

Comment:
It seems like `event.xdata` is sometimes a float and sometimes a tuple consisting of two floats. Any idea why?

Reply:
Because at least two places in `master` we do:
```
_draw_vert_line(np.array([event.xdata] * 2), params)
```
i.e., convert it to a list with two elements. But again this should not be necessary once you use `vlines` instead of `plot` for the vertical line drawing.